source/common/ssl/tls_certificate_config_impl.cc

```diff
@@ -45,12 +45,11 @@ std::vector<uint8_t> maybeReadOcspStaple(const envoy::config::core::v3::DataSour
 
 static const std::string INLINE_STRING = "<inline>";
 
-absl::StatusOr<std::unique_ptr<TlsCertificateConfigImpl>> TlsCertificateConfigImpl::create(
+absl::StatusOr<TlsCertificateConfigImpl> TlsCertificateConfigImpl::create(
     const envoy::extensions::transport_sockets::tls::v3::TlsCertificate& config,
     Server::Configuration::TransportSocketFactoryContext& factory_context, Api::Api& api) {
   absl::Status creation_status = absl::OkStatus();
-  std::unique_ptr<TlsCertificateConfigImpl> ret(
-      new TlsCertificateConfigImpl(config, factory_context, api, creation_status));
+  TlsCertificateConfigImpl ret(config, factory_context, api, creation_status);
```

Comment:
Does this type have a move constructor (to avoid extra copies)? It probably doesn't matter a lot because these are only moved while loading config, but if it's easy it might be nicer.

Reply:
I was under the impression that `TlsCertificateConfigImpl` will keep its default move c'tor (see full file here: https://github.com/envoyproxy/envoy/blob/db8f6e86ed23c66d149be2cc3b175d1ee3a53a19/source/common/ssl/tls_certificate_config_impl.h).

I've now explicitly added the default move c'tor in [3d9bb64](https://github.com/envoyproxy/envoy/pull/39270/commits/3d9bb64187251d1c050a241a682b1b6e4882d7b0) - just in case.
If you think it is not needed, I can remove it.